src/test/java/com/artipie/front/api/RepositoriesMoveTest.java

```diff
@@ -24,57 +32,68 @@
  * Test for {@link Repositories.Move}.
  * @since 0.1
  * @checkstyle ClassDataAbstractionCouplingCheck (500 lines)
+ * @checkstyle MagicNumberCheck (500 lines)
  */
 @SuppressWarnings("PMD.AvoidDuplicateLiterals")
 class RepositoriesMoveTest {
 
     /**
-     * Test repository settings.
+     * Temp dir.
+     * @checkstyle VisibilityModifierCheck (500 lines)
      */
-    private static final String REPO = String.join(
-        System.lineSeparator(),
-        "repo:",
-        "  type: rpm",
-        "  storage:",
-        "    type: fs",
-        "    path: /var/artipie/data/"
-    );
+    @TempDir
+    Path temp;
 
     /**
      * Test storage.
      */
     private BlockingStorage blsto;
 
+    /**
+     * Data storage.
+     */
+    private Storage asto;
+
     @BeforeEach
     void init() {
         this.blsto = new BlockingStorage(new InMemoryStorage());
+        this.asto = new FileStorage(this.temp);
     }
 
     @Test
-    void movesRepo() {
+    void movesRepo() throws InterruptedException {
         final var resp = Mockito.mock(Response.class);
         final var rqs = Mockito.mock(Request.class);
         final Key.From key = new Key.From("my-rpm.yml");
-        this.blsto.save(key, RepositoriesMoveTest.REPO.getBytes(StandardCharsets.UTF_8));
-        Mockito.when(rqs.params(Repositories.REPO_PARAM.toString())).thenReturn("my-rpm");
+        this.blsto.save(key, this.repoSettings().getBytes(StandardCharsets.UTF_8));
+        final String name = "my-rpm";
+        this.asto.save(new Key.From(name, "data"), Content.EMPTY).join();
+        Mockito.when(rqs.params(Repositories.REPO_PARAM.toString())).thenReturn(name);
         Mockito.when(rqs.attribute(RequestAttr.Standard.USER_ID.attrName())).thenReturn("any");
         Mockito.when(rqs.body()).thenReturn("{ \"new_name\": \"alice-rpm\"}");
+        final RepoSettings stn = new RepoSettings("flat", this.blsto);
         MatcherAssert.assertThat(
             "Failed to return empty response",
-            new Repositories.Move(new RepoSettings("flat", this.blsto)).handle(rqs, resp),
+            new Repositories.Move(stn, new RepoData(stn)).handle(rqs, resp),
             new IsAnything<>()
         );
         Mockito.verify(resp).status(HttpStatus.OK_200);
+        Thread.sleep(3000);
         MatcherAssert.assertThat(
             "Failed to read repository settings",
             new String(this.blsto.value(new Key.From("alice-rpm.yml")), StandardCharsets.UTF_8),
-            new IsEqual<>(RepositoriesMoveTest.REPO)
+            new IsEqual<>(this.repoSettings())
         );
         MatcherAssert.assertThat(
             "Repo with old name should not exist",
             this.blsto.exists(key),
             new IsEqual<>(false)
         );
+        MatcherAssert.assertThat(
+            "Data were moved",
+            this.asto.list(Key.ROOT).join().stream().map(Key::string).collect(Collectors.toList()),
+            Matchers.contains("alice-rpm/data")
```

Comment:
Should we check that 'my-rpm/data' does not exist?

Reply:
`Matchers.contains` is strict: it will fail, if collection contains some extra items, also using `ROOT` key we list all the items in the repository, which means, old repository does not exist.